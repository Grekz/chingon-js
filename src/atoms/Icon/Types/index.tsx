import Facebook from './Facebook'
import LinkedIn from './LinkedIn'
export default { Facebook, LinkedIn }
