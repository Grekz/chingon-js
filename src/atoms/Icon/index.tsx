// import { IconTypes } from './icon.component'
export { IconTypes } from './icon.component'
export { default } from './icon.component'
