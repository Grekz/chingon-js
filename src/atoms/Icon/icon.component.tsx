import * as React from 'react'
import Types from './Types'

interface IIconProps {
  size?: string
}

type IconComponent = React.SFC<IIconProps>

interface IIconComponentProps extends IIconProps {
  iconComponent: IconComponent
}

const Icon = ({ iconComponent, size }: IIconComponentProps) => {
  if (!iconComponent) {
    return null
  }
  const Component = iconComponent
  return <Component size={size || '1.5em'} />
}

export const IconTypes = Types
export default Icon
