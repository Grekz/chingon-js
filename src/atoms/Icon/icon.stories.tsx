import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Icon, { IconTypes } from '.'

storiesOf('Atoms|Icon', module).add('As Default', () => {
  return <Icon iconComponent={IconTypes.Facebook} size="100px" />
})
