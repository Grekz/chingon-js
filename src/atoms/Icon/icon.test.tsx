import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Icon, { IconTypes } from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Icon /> rendering', () => {
  it('should render Icon', () => {
    const wrapper = shallow(<Icon iconComponent={IconTypes.Facebook} />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<Icon iconComponent={IconTypes.Facebook} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
