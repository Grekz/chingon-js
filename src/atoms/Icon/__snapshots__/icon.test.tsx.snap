// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> rendering should match snapshot 1`] = `
<Facebook
  size="1.5em"
/>
`;
